src/config.c

```diff
@@ -2934,6 +2948,40 @@ standardConfig configs[] = {
     {NULL}
 };
 
+/* Create a new config by copying the passed in config. Returns 1 on success
+ * or 0 when their was already a config with the same name.. */
+int registerConfigValue(const char *name, const standardConfig *config, int alias) {
+    standardConfig *new = zmalloc(sizeof(standardConfig));
```

Comment:
I think a better idea to avoid creating separate standardConfig instances for the same config with alias is to have a wrapper around static standardConfig that takes the following shape

```
struct standardConfigWrapper {
    unsigned int flags;
    standardConfig *config;
};
```
what is inserted into the dict will be the new standardConfigWrapper rather standardConfig so this way you will have the following relationship

```
real_name-----> standardConfigWrapper----|
                                         V
                                         | ------> standardConfig
                                         ^
alias --------> standardConfigWrapper----|
```